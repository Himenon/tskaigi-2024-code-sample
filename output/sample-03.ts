const mainTask = () => {
    const subTask = () => {}
    return subTask();
  }
  mainTask();
  
  
