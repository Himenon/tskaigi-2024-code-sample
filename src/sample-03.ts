import { Project, ts } from "ts-morph";

export const run = () => {
  const text = `const mainTask = () => {
    const subTask = () => {}
    return subTask();
  }
  mainTask();
  
  `;

  const project = new Project();
  const sourceFile = project.createSourceFile("output/sample-03.ts", "", {
    overwrite: true,
  });

  sourceFile.addStatements([text]);

  project.saveSync();

  console.log("Generated output/sample-03.ts");
};
