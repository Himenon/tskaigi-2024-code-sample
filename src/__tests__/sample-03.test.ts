import * as fs from "node:fs";
import { test, expect } from "vitest";
import * as Module from "../sample-03";

test("Sample3: Code Generate Test", () => {
  Module.run();
  const generateCode = fs.readFileSync("output/sample3.ts", {
    encoding: "utf-8",
  });
  expect(generateCode).toMatchSnapshot();
});
