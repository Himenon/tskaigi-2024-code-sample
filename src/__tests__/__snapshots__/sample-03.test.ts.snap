// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sample3: Code Generate Test 1`] = `
"const mainTask = () => {
    const subTask = () => {}
    return subTask();
  }
  mainTask();
  
  
"
`;
